### DISCUSSION TITLE
Support more location types as cities?

### DISCUSSION THREAD

**Discussion by GoldJohnKing** [Vdauphin/HeartsAndMinds#1046] [Question and answer] [answered]
Hi.

I've recently noticed H&M only support `"NameVillage", "NameCity", "NameCityCapital", "NameLocal", "Hill", "Airport", "NameMarine"` as cities.

However, some of the workshop terrains, such as Napf and those made by IceBreakr, there are many other types of locations, such as:
```
Name
Flag
Strategic
StrongpointArea
BorderCrossing
VegetationBroadleaf
VegetationFir
VegetationPalm
VegetationVineyard
ViewPoint
```

Missing those types of locations makes H&M feels much more deserted on workshop terrains. (On Altis and Tanoa, cities (red circles) are much denser than those workshop terrains). I wonder if those types of location can be supported and regarded the same as `Hill`, which will make H&M feels much better on workshop terrains.

Also, `CulturalProperty` and `HistoricalSite` can also be regarded the same as `Church`.

References:
Location - BI Wiki: https://community.bistudio.com/wiki/Location

**Comment by Vdauphin** [comment, marked as answer]
Hello,

I see StrongpointArea interesting for Lingor island at least.
VegetationPalm have huge radius and may overlap with a lot of city. Even if this isn't a problem since 1.21, we don't want such huge area be activated.
ViewPoint is not a location from my point of view, just an interesting spot not related to building around etc

I think each types of locations should be choose carefully by looking how map author use: 
- do we have building around
- is there any location already supported just 100m or 200m around

> Also, CulturalProperty and HistoricalSite can also be regarded the same as Church.

Don't really understand this but CulturalProperty and HistoricalSite is not supported by https://community.bistudio.com/wiki/nearestTerrainObjects 
Churchs are used to center city when a location is in water but it isn't a NameMarine. This avoid spawning stuf in water.
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/init.sqf#L43

**Reply by GoldJohnKing** [reply]
> > Also, CulturalProperty and HistoricalSite can also be regarded the same as Church.
> 
> Don't really understand this but CulturalProperty and HistoricalSite is not supported by https://community.bistudio.com/wiki/nearestTerrainObjects
> Churchs are used to center city when a location is in water but it isn't a NameMarine. This avoid spawning stuf in water.
> https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/init.sqf#L43

Ah, I mixed up those things...

**Comment by GoldJohnKing** [comment]
> I see StrongpointArea interesting for Lingor island at least.

Yes, I see `StrongpointArea` and `BorderCrossing` are both widely used in workshop terrains. Both of these represents an important strategic area on a terrain, so I wonder if both of them can be supported.

`BorderCrossing` mostly represents checkpoint on road which connect two regions/countries. You can see this in many workshop terrains, for example, IceBreakr's Fapovo Island.

> VegetationPalm have huge radius and may overlap with a lot of city. Even if this isn't a problem since 1.21, we don't want such huge area be activated.

In many workshop terrains with a large scale, there are many wide areas of forest and vegetations, which are far from the cities and far from roads. These areas are very good places for resistance forces like "Oplitas" to hide, but now they are pretty deserted and unpopulated.

As the max city radius is limited and should not be a problem, I think it would be better to have those vegetation areas (`VegetationBroadleaf`, `VegetationFir`, `VegetationPalm`, `VegetationVineyard`), too.

> ViewPoint is not a location from my point of view, just an interesting spot not related to building around etc

As for `ViewPoint`, they are mostly high areas with wide range of sights, should be good places where resistance would like to occupy and civilians would like to visit. Should be regarded the same as `Hill`, because most of them share the same style.

Besides, in many workshop terrains (especially Lingor and Fapovo), many locations that would be regarded as cities in H&M, are much too overlap. I often see two city center within the each city's radius, or even closer.

This could be an issue. Players a area that cities are too close to each other, can always activate too much cities at once and cause H&M spawn too much units, that will cause these areas has much stronger enemy strength, and even worse, the server overload.

I wonder if we could handle this differently: once two cities are in each one's radius, regard them as one city and increase it's radius according to priority.

For example, `Airport` > `NameCityCapital` > `NameCity`, so if an `Airport` and a `NameCityCapital` is in each other's range, `NameCityCapital` should not be regarded as a single city anymore, while the `Airport` will have a larger radius according to the `NameCityCapital`'s radius.

We can also have a formula on the radius calculation, for example, `(city1_radius + city2_radius + activation_radius)`.

I think this could better prevent the issue above.

**Reply by Vdauphin** [reply]
`StrongpointArea` and `BorderCrossing` really fit the criteria for a location handle by hearts and minds
Same for `VegetationFir` in fapovo, those vegetation have some building
I still don't like `VegetationPalm` as they have no buildings around and have big area even if max city radius is limited. It looks like `hill` but most of the time they cover multiple city
`VegetationBroadleaf` and `VegetationVineyard` don't see any use case

> As for ViewPoint, they are mostly high areas with wide range of sights, should be good places where resistance would like to occupy and civilians would like to visit. Should be regarded the same as Hill, because most of them share the same style.

I quickly tested lingor and fapovo (thanks to my Shadow Tech PC) but I found only see `ViewPoint` once in lingor and it looks pointless

> I wonder if we could handle this differently: once two cities are in each one's radius, regard them as one city and increase it's radius according to priority.

I really don't want to touch that. I want to keep stuff simple so simpler to debug

From what I saw `StrongpointArea`, `BorderCrossing` and `VegetationFir` fit location in city
